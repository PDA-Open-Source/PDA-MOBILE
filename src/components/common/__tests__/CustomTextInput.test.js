import React from 'react';
import renderer from 'react-test-renderer';
import CustomTextInput from '../CustomTextInput';
jest.mock('react-native-device-info');
describe('<TextInput/>',()=>{
  beforeEach(()=>{
  });
  it('should render the component',()=>{
    const snap = renderer.create(<CustomTextInput/>);
    expect(snap.toJSON()).toMatchSnapshot();
  });
  it('function focus should call the TextInput to focus',()=>{
    // wrapper.instance().focus(); @TODO
  });
});