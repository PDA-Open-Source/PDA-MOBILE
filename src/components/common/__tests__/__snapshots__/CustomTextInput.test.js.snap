// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput/> should render the component 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#ffffff",
        "flexDirection": "row",
        "marginVertical": 19.61764705882353,
      },
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "white",
        "flexDirection": "row",
        "marginVertical": 19.61764705882353,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 15,
        },
        Object {
          "alignItems": "center",
          "padding": 5,
          "paddingHorizontal": 10,
        },
        Object {
          "fontFamily": "Font Awesome 5 Pro",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "100",
        },
      ]
    }
  >
    
  </Text>
  <TextInput
    allowFontScaling={true}
    maxLength={250}
    multiline={false}
    numberOfLines={1}
    onChangeText={[Function]}
    onEndEditing={[Function]}
    onFocus={[Function]}
    onSubmitEditing={[Function]}
    placeholderTextColor="rgba(255, 255, 255, 0.55)"
    style={
      Array [
        Object {
          "color": "#ffffff",
          "flex": 1,
        },
        Object {
          "fontFamily": "Roboto-Regular",
          "fontSize": 23.57142857142857,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontFamily": "Roboto-Light",
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`<TextInput/> should render the compontent 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#fff",
        "flexDirection": "row",
        "marginVertical": 19.61764705882353,
      },
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "white",
        "flexDirection": "row",
        "marginVertical": 19.61764705882353,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 15,
        },
        Object {
          "alignItems": "center",
          "padding": 5,
          "paddingHorizontal": 10,
        },
        Object {
          "fontFamily": "Font Awesome 5 Pro",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontWeight": "100",
        },
      ]
    }
  >
    
  </Text>
  <TextInput
    allowFontScaling={true}
    maxLength={250}
    multiline={false}
    numberOfLines={1}
    onChangeText={[Function]}
    onEndEditing={[Function]}
    onFocus={[Function]}
    onSubmitEditing={[Function]}
    placeholderTextColor="rgba(255, 255, 255, 0.55)"
    style={
      Array [
        Object {
          "color": "#fff",
          "flex": 1,
        },
        Object {
          "fontFamily": "Roboto-Regular",
          "fontSize": 23.57142857142857,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {
          "fontFamily": "Roboto-Light",
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
